src/abstractdataframe/abstractdataframe.jl

```diff
@@ -375,29 +375,30 @@ describe(df, stats = [:min, :max])
 ```
 
 """
-function StatsBase.describe(df::AbstractDataFrame; stats::Union{Symbol,AbstractVector{Symbol}} =
-                            [:mean, :min, :median, :max, :nunique, :nmissing, :eltype])
-    # Check that people don't specify the wrong fields.
-    allowed_fields = [:mean, :std, :min, :q25, :median, :q75,
-                      :max, :nunique, :nmissing, :first, :last, :eltype]
-    if stats == :all
-        stats = allowed_fields
+StatsBase.describe(df::AbstractDataFrame) = describe(df, :mean, :min, :median, 
+                                                     :max, :nunique, :nmissing, 
+                                                     :eltype)
+function StatsBase.describe(df::AbstractDataFrame, stats::Union{Symbol, Pair}...)
```

Comment:
Probably it would be better to write `Union{Symbol, Pair{Symbol}}`?

Reply:
Also note my previous remark about the implementation to avoid mostly useless specialization (second paragraph): https://github.com/JuliaData/DataFrames.jl/pull/1664#issuecomment-452073893

Reply:
does this mean I should add another function barrier `_describe` that is `_describe(df::AbstractDataFrame, Vector{<:Union{Symbol, Pair{Symbol}})`?

Reply:
Yes.